password.py

```diff
@@ -1,49 +1,302 @@
 def account_save():
 
-    return
+    account_file = open('account.txt','w')
+    account_file.write(str(account_dict))
+    account_file.close()
+
+    return account_dict
+
+#lists account dict
+# def list_account():
+#     account_file_read = open('account.txt','r')
+#     print(account_file_read.read())
+#     account_file_read.close()
+
 # Recover exisiting account
 def account_recovery():
+    account_dict_file = open('account.txt','r')
+
+    for line in account_dict_file:
+        word = re.split('\{|: |, |\}', line)
+        name = word[1]
+        username = name.replace("'","")
+        pass_w = word[6]
+        password = pass_w.replace("'","")
+        sec_ques = word[8]
+        security_question = sec_ques.replace('"',"")
+        sec_answ = word[10]
+        security_answer = sec_answ.replace("'","")
+    
+    attempts = 4
+
+    while True:
+        print("Enter a username: ")
+        user_answer = input("")
+
+        if user_answer == username:
+            while True:
+                print(security_question)
+                question_answer = input("")
+                if question_answer == security_answer:
+                    print("Your password is: "+password)
+                    break
+                elif attempts == 0:
+                    print("You have exceeded max attempts.")
+                    break
+                else:
+                    attempts -= 1
+                    print("You have "+ str(attempts) +" attempt(s) left.")        
+            break
+        else:
+            print("Username not found!")
+            continue
+
+    account_dict_file.close()
 
     return
 
 #<---- SECOND MAJOR FUNCTION ---->
 # Login to existing account
 def login():
+    account_dict_file = open('account.txt','r')
 
+    attempts = 4
+    for line in account_dict_file:
+        word = re.split('\{|: |, |\}', line)
+        name = word[1]
+        username = name.replace("'","")
+        pass_w = word[6]
+        password = pass_w.replace("'","")
+
+    while True:
+        print("Enter your username: ")
+        user_input_username = input("")
+
+        if user_input_username == username:
+            while True:
+                print("Enter your password: ")
+                user_input_password = input("")
+                if user_input_password == password:
+                    password_manager()
+                    break
+                elif attempts == 0:
+                    print("You have exceeded max password attempts!")
+                    break
+                else:
+                    attempts -= 1
+                    print("Password not found!")
+                    print("You have "+ str(attempts) +" attempt(s) left.")
+            break
+
+        else:
+            print("Username not found!")
+            continue
+
+    account_dict_file.close()
+    
     return
+    
+#<---- THIRD MAJOR FUNCTION ---->
+# Manages and Rates your passwords.
+def password_manager():
```

Comment:
I'm kinda confused on the purpose of this? Is it a replacement for account create? It seems like they share a lot of the same code.

Reply:
I was debating putting this function in an entirely new file since the 'Account' process and the actual 'Password Managing" process were two completely different _ideas_ you could say.

**Account creating** and **Password managing** does have the same code, as they both are supposed to save inputs, with the difference being what they actually save at the end.
(in terms of **account** would save an entire account, and **password** just saves your password and maybe what that password is for in the future)

I think I subconsciously tried to make the function have more purpose here by fleshing out the password fail parameters more than the account password's fail parameters.

Reply:
I don't think you really need it to be its own separate logic train. The account function already does most of this stuff, so really all you need is to call distill the part of the program that changes the password and you'd have this. So instead maybe your program can instead give choices for this at the start: 

1) Create new entry
2) Change current entry

Then you can query the user for what they want to change and then call the appropriate function to take user input and update it.